File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        for(int i=0; i<commands.length; i++) {
            int copyArr[] = getArray(commands[i][0], commands[i][1], array);
            Arrays.sort(copyArr);
            answer[i] = copyArr[commands[i][2]-1];
        }
        
        return answer;
    }
    
    public int[] getArray(int start, int end, int target[]) {
        int arr[] = new int[end - start + 1];
        
        for(int i=0; i<end - start + 1; i++) {
            arr[i] = target[start - 1 + i];
        }
        return arr;
    }
}

File: 프로그래머스/lv1/42748. K번째수/K번째수.kt

class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = mutableListOf<Int>()

        for(i in 0 until commands.size){
            var sort = array.slice(commands[i][0]-1 .. commands[i][1]-1)
            answer.add(sort.sorted()[commands[i][2]-1])
        }
        return answer.toIntArray()
    }
}